cognite/client/data_classes/simulators/routines.py

```diff
@@ -0,0 +1,141 @@
+        external_id: str,
+        model_external_id: str,
+        simulator_integration_external_id: str,
+        name: str,
+        description: str | None = None,
+    ) -> None:
+        self.external_id = external_id
+        self.model_external_id = model_external_id
+        self.simulator_integration_external_id = simulator_integration_external_id
+        self.name = name
+        self.description = description
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        return cls(
+            external_id=resource["externalId"],
+            model_external_id=resource["modelExternalId"],
+            simulator_integration_external_id=resource["simulatorIntegrationExternalId"],
+            name=resource["name"],
+            description=resource.get("description"),
+        )
+
+
+class SimulatorRoutineWrite(SimulatorRoutineCore):
+    def as_write(self) -> SimulatorRoutineWrite:
+        """Returns a writeable version of this resource"""
+        return self
+
+
+class SimulatorRoutine(SimulatorRoutineCore):
+    def __init__(
+        self,
+        external_id: str,
+        model_external_id: str,
+        simulator_integration_external_id: str,
+        name: str,
+        id: int,
+        data_set_id: int,
+        simulator_external_id: str,
+        created_time: int,
+        last_updated_time: int,
+        description: str | None = None,
+    ) -> None:
+        super().__init__(
+            external_id=external_id,
+            model_external_id=model_external_id,
+            simulator_integration_external_id=simulator_integration_external_id,
+            name=name,
+            description=description,
+        )
+        # id/created_time/last_updated_time are required when using the class to read,
+        # but don't make sense passing in when creating a new object. So in order to make the typing
+        # correct here (i.e. int and not Optional[int]), we force the type to be int rather than
+        # Optional[int].
+        self.id = id
+        self.created_time = created_time
+        self.last_updated_time = last_updated_time
+        self.simulator_external_id = simulator_external_id
+        self.data_set_id = data_set_id
+
+    @classmethod
+    def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None = None) -> Self:
+        instance = cls(
+            external_id=resource["externalId"],
+            simulator_external_id=resource["simulatorExternalId"],
+            model_external_id=resource["modelExternalId"],
+            simulator_integration_external_id=resource["simulatorIntegrationExternalId"],
+            name=resource["name"],
+            data_set_id=resource["dataSetId"],
+            description=resource.get("description"),
+            created_time=resource["createdTime"],
+            last_updated_time=resource["lastUpdatedTime"],
+            id=resource["id"],
+        )
+        return instance
```

Comment:
My personal preference, up to you to follow it.
```suggestion
        return cls(
            external_id=resource["externalId"],
            simulator_external_id=resource["simulatorExternalId"],
            model_external_id=resource["modelExternalId"],
            simulator_integration_external_id=resource["simulatorIntegrationExternalId"],
            name=resource["name"],
            data_set_id=resource["dataSetId"],
            description=resource.get("description"),
            created_time=resource["createdTime"],
            last_updated_time=resource["lastUpdatedTime"],
            id=resource["id"],
        )
```